Add support for Xamarin.Watch profile.

rolfbjarne opened the pull request:
Adds support for the upcoming Xamarin.Watch profile (for the Apple Watch) in Xamarin.iOS.

It's build whenever the `monotouch` profile is also built, since we'll never build them separately anyway.

dsyme commented on config.make.in:
```diff
@@ -60,6 +60,11 @@ VERSION = 2.3.98.1
 endif
 
+ifeq (x-$(TargetFramework)-$(FSharpCoreBackVersion),x-xamarinwatch-)
+VERSION = 2.3.98.1
```
monotouch and monodroid share the same version number because, as far as @7sharp9 and others could determine, the FSharp.Core for these two is identical and reference identically versioned mscorlib etc.

What do we know about Xamarin.Watch?  Looking at the list below it looks like it would be safer and more consistent to give it a platform version number like 2.3.101.1.

rolfbjarne replied:
It sounds safer, but I don't know what kind of problems different version numbers can cause.

FWIW xamarinwatchos is be closer to monotouch than monodroid is, which is why I went with the same version number in the first place.

rolfbjarne commented:
I've now rebased this branch on top of master (there was a merge failure, which I fixed), and I've renamed "Xamarin.Watch" to "Xamarin.WatchOS", since that's the name we decided to go with.

dsyme commented:
OK, this looks good.  

Note that my understanding from the FSharpSource.targets defines (which are identical) is that the FSharp.Core being built for Xamarin.WatchOS is identical to that for monodroid and monotouch.  The only real reason we distinguish between monodroid and monotouch at all is because we can't be sure which of the two are installed at build time.

Because the same version numbers are being used for these different FSharp.Core, we have to be careful going forward that this assumption doesn't change.
